File: src/main/java/com/pubnub/api/models/consumer/pubsub/MessageResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString(callSuper = true)
public class MessageResult extends BasePubSubResult {

    private JsonElement message;

    public MessageResult(BasePubSubResult basePubSubResult, JsonElement message) {
        super(basePubSubResult);
        this.message = message;
    }
}

File: src/main/kotlin/com/pubnub/api/models/consumer/pubsub/MessageResult.kt
package com.pubnub.api.models.consumer.pubsub

import com.google.gson.JsonElement
import com.pubnub.api.SpaceId

/**
 * @property message The actual message content
 */
interface MessageResult : PubSubResult {
    val message: JsonElement
    val spaceId: SpaceId?
    val type: String?
}
